document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -283,17 +284,51 @@ public boolean updateSubDoc(Key key, String subDocPath, Document subDocument) {
       UpdateResult writeResult =
           collection.updateOne(selectionCriteriaForKey(key), setObject, new UpdateOptions());
       if (LOGGER.isDebugEnabled()) {
-        LOGGER.debug("Write result: " + writeResult.toString());
+        LOGGER.debug("Write result: " + writeResult);
       }
       // TODO:look into the writeResult to ensure it was successful. Was not easy to find this from
       // docs.
       return true;
     } catch (IOException e) {
-      LOGGER.error("Exception inserting document. key: {} content:{}", key, subDocument);
+      LOGGER.error("Exception updating document. key: {} content:{}", key, subDocument);
       return false;
     }
   }
 
+  @Override
+  public boolean bulkUpdateSubDoc(Map<Key, Map<String, Document>> bulkUpdate) {
+    List<UpdateManyModel<BasicDBObject>> bulkWriteUpdate = new ArrayList<>();
+    for (Key key : bulkUpdate.keySet()) {
+      Map<String, Document> entityUpdateOperations = bulkUpdate.get(key);
```

Comment:
We shouldn't use the term entity here.

```suggestion
      Map<String, Document> subDocuments = bulkUpdate.get(key);
```